docker/chain/Dockerfile

```diff
@@ -1,63 +1,72 @@
-RUN echo "echo '- Fund admin'" >> reset
-RUN echo "./fund-admin" >> reset
-RUN echo "echo '- Done'" >> reset
-RUN chmod +x reset
 
+RUN apk update
+RUN apk add bash
+RUN apk add nodejs
+RUN apk add npm
+
+WORKDIR /l2chain
+COPY . .
 RUN touch run
-RUN echo "#!/bin/bash" >> run
+RUN echo "dockerd-entrypoint.sh & " >> run
+RUN echo "until docker info >/dev/null 2>&1; do " >> run
+RUN echo "   echo 'Waiting for Docker to start...'" >> run
+RUN echo "   sleep 1" >> run
+RUN echo "done" >> run
+
+
+RUN echo "docker pull offchainlabs/nitro-node:v2.3.4-b4cc111 " >> run
 
-FROM base AS expose-cloudflare
-RUN curl -L --output cloudflared.deb https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-${arch}.deb
-RUN dpkg -i cloudflared.deb
-RUN echo "cloudflared tunnel --metrics 0.0.0.0:11111 run --token $cloudflare_token_http --url http://localhost:8545 &" >> run
-RUN echo "cloudflared tunnel --metrics 0.0.0.0:11112 run --token $cloudflare_token_ws --url http://localhost:8546 &" >> run
+RUN echo "docker run -d --name devnet -p 0.0.0.0:8547:8547 -p 0.0.0.0:8548:8548 offchainlabs/nitro-node:v2.3.4-b4cc111  "\
+"   --init.dev-init " \
+"   --init.dev-init-address 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 "\
+"   --node.dangerous.no-l1-listener "\
+"   --node.parent-chain-reader.enable=false "\
+"   --parent-chain.id=1337 "\
+"   --chain.id=412346 "\
+"   --persistent.chain /tmp/dev-test "\
+"   --node.sequencer "\
+"   --execution.sequencer.enable "\
+"   --node.dangerous.no-sequencer-coordinator "\
+"   --node.staker.enable=false "\
+"   --init.empty=false "\
+"   --http.port 8547  "\
+"   --http.addr 0.0.0.0 "\
+"   --ws.port 8548  "\
+"   --ws.addr 0.0.0.0 "\
+"   --http.corsdomain=* "\
+"   --http.vhosts=* "\
+"   --ws.origins=* "\
+"    " >> run
 
-FROM base AS expose-local
-EXPOSE 8545
-EXPOSE 8546
+RUN echo "until wget -qO- http://0.0.0.0:8547 >/dev/null 2>&1; do " >> run
+RUN echo "   echo 'Waiting for Nitro to start...'" >> run
+RUN echo "   sleep 1" >> run
+RUN echo "done" >> run
 
-FROM expose-$expose_via AS final
-RUN echo "./run-node" >> run
-RUN echo "sleep infinity" >> run
+RUN echo "cd hardhat" >> run
+RUN echo "npm install" >> run
+RUN echo "npx hardhat deploy --network devnet" >> run
+RUN echo "npx hardhat run scripts/fund-services-ether.ts  --network devnet" >> run
+RUN echo "npx hardhat run scripts/fund-services-tokens.ts  --network devnet " >> run
+RUN echo "npx hardhat run scripts/print-l2-config.ts  --network devnet" >> run
+
+RUN wget https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-amd64
+RUN echo "cd .." >> run
+RUN echo "./cloudflared-linux-amd64 tunnel run --token $cloudflare_token_http  &" >> run
+RUN echo "./cloudflared-linux-amd64 tunnel run --token $cloudflare_token_ws &" >> run
+
+RUN echo "tail -f /dev/null" >> run
+
+RUN chmod +x cloudflared-linux-amd64
 RUN chmod +x run
 
-CMD ["/bin/bash", "./run"]
+ENTRYPOINT ["bash", "./run"]  
+
+#example use:
```

Comment:
Can you please add two commands to `stack`? One for building and one for running for local purposes.

Reply:
And can you also add the `expose_via` mecahanism? This way, we can test the containerized version locally without having to setup cloudflare tunnels.